examples_tests/tests.py

```diff
@@ -0,0 +1,250 @@
+            'version': '1',
+            }),
+        ('ros', {
+            'dir': 'ros',
+            'name': 'ros-example',
+            'version': '1.0',
+            }),
+        ('shout', {
+            'dir': 'shout',
+            'name': 'shout',
+            'version': '0.52.0',
+            }),
+        ('tomcat-maven-webapp', {
+            'dir': 'tomcat-maven-webapp',
+            'name': 'tomcat-webapp-demo',
+            'version': '1.0',
+            }),
+        ('webcam-webui', {
+            'dir': 'webcam-webui',
+            'name': 'webcam-webui',
+            'version': '1',
+            }),
+        ('webchat', {
+            'dir': 'webchat',
+            'name': 'webchat',
+            'version': '0.0.1',
+            }),
+    ]
+
+    @classmethod
+    def setUpClass(cls):
+        cls.temp_dir = tempfile.mkdtemp()
+        if not config.get('skip-install', False):
+            ip = config.get('ip', None)
+            if ip is None:
+                cls.testbed_ip = 'localhost'
+                cls.testbed_port = '8022'
+                cls.vm_process = _start_snappy_testbed(
+                    cls.temp_dir, cls.testbed_port)
+            else:
+                cls.testbed_ip = ip
+                cls.testbed_port = config.get('port', '22')
+
+            _wait_for_ssh(cls.testbed_ip, cls.testbed_port)
+
+    @classmethod
+    def tearDownClass(cls):
+        if cls.vm_process:
+            cls.vm_process.kill()
+        shutil.rmtree(cls.temp_dir)
+
+    def run_command_through_ssh(self, command):
+        return _run_command_through_ssh(
+            self.testbed_ip, self.testbed_port, command)
+
+    def build_snap(self, project_dir):
+        snapcraft = os.path.join(os.getcwd(), 'bin/snapcraft')
+        subprocess.check_call([snapcraft, 'clean'], cwd=project_dir)
+        subprocess.check_call(snapcraft, cwd=project_dir)
+
+    def copy_snap_to_testbed(self, snap_local_path):
+        _copy_file_through_ssh(
+            self.testbed_ip, self.testbed_port, 'ubuntu',
+            snap_local_path, '/home/ubuntu/')
+
+    def delete_snap_from_testbed(self, snap_file_name):
+        self.run_command_through_ssh(
+            ['rm', os.path.join('/home/ubuntu/', snap_file_name)])
+
+    def install_snap(self, snap_file_name):
+        self.run_command_through_ssh(
+            ['sudo', 'snappy', 'install', snap_file_name])
+
+    def remove_snap(self, snap_name):
+        self.run_command_through_ssh(['sudo', 'snappy', 'remove', snap_name])
+
+    def test_example(self):
+        if self.name == 'qmldemo' and _get_ubuntu_version() == 'trusty':
+            self.skipTest('qmldemo is not supported on trusty.')
+
+        example_dir = os.path.join('examples', self.dir)
+        # Build snap will raise an exception in case of error.
+        self.build_snap(example_dir)
+
+        if not config.get('skip-install', False):
+            snap_file_name = '{}_{}_amd64.snap'.format(self.name, self.version)
```

Comment:
`amd64` needs to probably change so we can eventually do all this on `armhf` or `arm64`